include/dlaf/matrix/distribution.h

```diff
@@ -119,6 +119,17 @@ class Distribution {
 
   Distribution& operator=(Distribution&& rhs) noexcept;
 
+  /// Constructs a sub-distribution based on the given distribution @p dist with
+  /// an @p offset and @p size.
+  ///
+  /// @param[in] dist is the input distribution,
+  /// @param[in] offset is the offset of the new distribution relative to the input distribution,
+  /// @param[in] size is the size of the new distribution relative to the offset,
+  /// @pre origin.isValid()
+  /// @pre size.isValid()
+  /// @pre origin + size <= dist.size()
+  Distribution(Distribution dist, const GlobalElementIndex& offset, const GlobalElementSize& size);
```

Comment:
This currently takes `offset` and `size` as separate parameters. Some alternatives:

- Add a `SubDistributionSpec` (similar to `SubTileSpec`) and use that instead. I would not name it `SubMatrixSpec` here since we don't want to make `Distribution` "depend" on `Matrix` (it wouldn't be a real dependency, but a conceptual dependency). We could perhaps add a `using SubMatrixSpec = SubDistributionSpec` to `matrix.h` as well.
- Allow defaulting the `size` to use the rest of the distribution/matrix. With separate parameters this is trivial (e.g. `const std::optional<GlobalElementSize>& size = std::nullopt`). With a `SubDistributionSpec` it could be done either with an `optional` member in `SubDistributionSpec` or two constructors: `Distribution(Distribution dist, const GlobalElementIndex& offset)` and `Distribution(Distribution dist, const SubDistributionSpec& spec)`. The latter would probably be cleaner.

Thoughts?

Reply:
I would be a fan of the first solution with `SubDistributionSpec` (and in case of need adding the aliasing as you suggested for `SubMatrixSpec`).

Reply:
I've added
```
struct SubDistributionSpec {
    GlobalElementIndex origin;
    GlobalElementSize size;
};
```
and
```
using SubMatrixSpec = SubDistributionSpec;`
```
This lead to some `offset` to `origin` renamings as well. I'm not 100% confident that I've been consistent in using `origin` vs. `offset` (or that I even have a consistent rule to follow) so I'd ask you to keep an eye out and see if some namings don't make sense even after we merge this.